internal/services/msi/user_assigned_identity_resource.go

```diff
@@ -118,7 +118,7 @@ func resourceArmUserAssignedIdentityRead(d *pluginsdk.ResourceData, meta interfa
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := commonids.ParseUserAssignedIdentityID(d.Id())
+	id, err := commonids.ParseUserAssignedIdentityIDInsensitively(d.Id())
```

Comment:
I think we'll need to do a state migration here to ensure the ID stored in state is correct? like we did in #18291

Reply:
d.id is the resource ID? so much like #18291 we'll need to apply the same type of fix including a state migration for existing users insuring that the ID in state is in the correct case

Reply:
Ahh, I'm with stupid. I'll fix tomorrow. Next release then.. or the former 2 without this one. Not at my pc atm unfortunately

Reply:
it happens! i'll remove this bit from this PR so we can get the other two into todays release and leave the user assigned identity for next

Reply:
```suggestion
	id, err := commonids.ParseUserAssignedIdentityID(d.Id())
```